### PR TITLE
Add support for config remote loading

### PR DESCRIPTION
**Pull request by victorboissiere**
# Description

You can now load the config from remote source to support new use cases like global config within a company.

Usage:
```
# Remote
go run . run --config=https://mylocalconfig/config.yaml
# Locally
go run . run --config=config.yaml
```

### CODE REVIEW

**Comment by victorboissiere** [comment]
#sre cc @mcorbin

File: config/utils.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+)
+
+func getConfigYaml(path string) ([]byte, error) {
+	if strings.HasPrefix(path, "http") {
+		res, err := http.Get(path)
+		if err != nil {
+			return []byte{}, err
+		}
+
+		if res.StatusCode != 200 {
```
Comment (airclovis) [inline config/utils.go:19]: You can use http status from std lib here like:
```suggestion
		if res.StatusCode != http. StatusOK {
```

File: config/utils.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+			return []byte{}, errors.New(fmt.Sprintf("Request returned: %d", res.StatusCode))
+		}
+
+		resBody, err := ioutil.ReadAll(res.Body)
```
Comment (airclovis) [inline config/utils.go:23]: ioutil is deprecated see [here](https://pkg.go.dev/io/ioutil#ReadAll)

You shuold use 
```suggestion
		resBody, err := io.ReadAll(res.Body)
```

File: config/utils.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	"strings"
+)
+
+func getConfigYaml(path string) ([]byte, error) {
+	if strings.HasPrefix(path, "http") {
+		res, err := http.Get(path)
+		if err != nil {
+			return []byte{}, err
+		}
+
```
Comment (airclovis) [inline config/utils.go:18]: If there's no error doing the `Get` call, we can add a defer function to close the response body
```suggestion
defer res.Body.Close()
```

File: config/utils.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+	"net/http"
+	"os"
+	"strings"
+)
+
+func getConfigYaml(path string) ([]byte, error) {
+	if strings.HasPrefix(path, "http") {
+		res, err := http.Get(path)
+		if err != nil {
+			return []byte{}, err
```
Comment (airclovis) [inline config/utils.go:16]: `nil` is a valid value for `[]byte` type 🙂 You can use nil in the returns in case of error to avoid one allocation

**Review by mcorbin — APPROVED** [review]

**Pull request merged**